File: src/lib/types/execution.ts
export type ErrorCode =
  | "no_active_prompt"
  | "execution_in_progress"
  | "clipboard_error"
  | "api_error"
  | "validation_error"
  | "timeout_error"
  | "unknown_error";

export interface ExecutionResult {
  success: boolean;
  content: string | null;
  error: string | null;
  error_code: ErrorCode | null;
  execution_time: number | null;
  metadata: unknown | null;
  execution_id: string | null;
}

File: src-tauri/src/models/execution.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ErrorCode {
    NoActivePrompt,
    ExecutionInProgress,
    ClipboardError,
    ApiError,
    ValidationError,
    TimeoutError,
    UnknownError,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ExecutionResult {
    pub success: bool,
    #[serde(default)]
    pub content: Option<String>,
    #[serde(default)]
    pub error: Option<String>,
    #[serde(default)]
    pub error_code: Option<ErrorCode>,
    #[serde(default)]
    pub execution_time: Option<f64>,
    #[serde(default)]
    pub metadata: Option<serde_json::Value>,
    #[serde(default)]
    pub execution_id: Option<String>,
}
